[BUG] rdiff-backup fails to build with Python 3.13: Missing <unistd.h> declaration

## Bug summary

rdiff-backup fails to build with Python 3.13.0a1.

## Version, Python, Operating System

rdiff-backup-2.2.6

## rdiff-backup call

Test mass rebuild of all packages with Python 3.13.

## What happened and what did you expect?

rdiff-backup fails to build with Python 3.13.0a1.

```
src/_librsyncmodule.c: In function ‘_librsync_new_patchmaker’:
src/_librsyncmodule.c:407:16: error: implicit declaration of function ‘dup’ [-Werror=implicit-function-declaration]
  407 |   int dup_fd = dup(python_fd);
      |                ^~~
```

## More information

From [Red Hat Bugzilla – Bug 2245885](https://bugzilla.redhat.com/show_bug.cgi?id=2245885)

According to [https://docs.python.org/3.13/whatsnew/3.13.html](https://docs.python.org/3.13/whatsnew/3.13.html):

Python.h no longer includes the <unistd.h> standard header file.
If needed, it should now be included explicitly.
For example, it provides the functions: read(), write(), close(), isatty(), lseek(), getpid(), getcwd(), sysconf() and getpagesize().
As a consequence, _POSIX_SEMAPHORES and _POSIX_THREADS macros are no longer defined by Python.h.
The HAVE_UNISTD_H and HAVE_PTHREAD_H macros defined by Python.h can be used to decide if <unistd.h> and <pthread.h> header files can be included.
(Contributed by Victor Stinner in gh-108765.)

For the build logs, see:
[https://copr-be.cloud.fedoraproject.org/results/@python/python3.13/fedora-rawhide-x86_64/06546852-rdiff-backup/](https://copr-be.cloud.fedoraproject.org/results/@python/python3.13/fedora-rawhide-x86_64/06546852-rdiff-backup/)

For all our attempts to build rdiff-backup with Python 3.13, see:
[https://copr.fedorainfracloud.org/coprs/g/python/python3.13/package/rdiff-backup/](https://copr.fedorainfracloud.org/coprs/g/python/python3.13/package/rdiff-backup/)

Testing and mass rebuild of packages is happening in copr.
You can follow these instructions to test locally in mock if your package builds with Python 3.13:
[https://copr.fedorainfracloud.org/coprs/g/python/python3.13/](https://copr.fedorainfracloud.org/coprs/g/python/python3.13/)

Let us know here if you have any questions.

Python 3.13 is planned to be included in Fedora 41.
To make that update smoother, we're building Fedora packages with all pre-releases of Python 3.13.
A build failure prevents us from testing all dependent packages (transitive [Build]Requires),
so if this package is required a lot, it's important for us to get it fixed soon.

We'd appreciate help from the people who know this package best,
but if you don't want to work on this now, let us know so we can try to work around it on our side.

--- a/src/_librsyncmodule.c
+++ b/src/_librsyncmodule.c
@@ -35,6 +35,15 @@ static inline void _Py_SET_TYPE(PyObject *ob, PyTypeObject *type)
 #define Py_SET_TYPE(ob, type) _Py_SET_TYPE((PyObject*)(ob), type)
 #endif
 
+/* ----------------------------------------------------------------------- *
+ * Update for Python 3.13/3.14 - Contributed by Victor Stinner in gh-108765
+ * https://docs.python.org/3.13/whatsnew/3.13.html#id8 - see issue #934
+ * The change was retracted but might come back in 3.14 so we keep this fix.
+ * ----------------------------------------------------------------------- */
+#if (defined(HAVE_UNISTD_H) && !defined(_UNISTD_H))
+#  include <unistd.h>
+#endif
+
 static PyObject *librsyncError;
 
 /* Sets python error string from result */

